src/serlo-editor/plugins/text/hooks/use-editable-paste-handler.tsx

```diff
@@ -39,9 +38,8 @@ export const useEditablePasteHandler = (args: UseEditablePasteHandlerArgs) => {
       if (!files.length && !text) return
 
       // Iterate through all plugins and try to process clipboard data
-      const plugins = editorPlugins.getAllWithData()
       let media
-      for (const { plugin, type } of plugins) {
+      for (const { plugin, type } of editorPlugins.getAllWithData()) {
```

Comment:
I prefer keeping it as it is, because it will exit as soon as it processes the clipboard data, while the suggestion will map all of them, and then only exit once it `find`s.

>it's a Jonas legacy that we use functional version wherever possible in the codebase

I love functional when it makes sense, but using functional, only for the sake of using functional, is one of my least favorite things about functional.

Reply:
that's a pretty functional view of functional I guess ;)